src/writer/_dirent.h

```diff
@@ -142,7 +142,7 @@ namespace zim
         DirentInfo info;
         offset_t offset;
         uint8_t _ns : 2;
-        bool removed;
+        bool removed : 1;
 
```

Comment:
My (probably wrong) belief/understanding was that adjacent bit-fields are packed only if they have the same type, but I cannot find confirmation of it (they are not even *required* to be packed; instead it is an expected implementation-defined behaviour).

Reply:
I've just tested and the sizeof the dirent is 38B whatever we use a bit-field with same type or not (or a simple plain uint8_t).
But your belief seems to be correct (https://stackoverflow.com/questions/308364/c-bitfield-packing-with-bools).
I've added a unit test for the dirent size. If we found some compiler are not packing the bitfield correctly we will change the code.

Reply:
It is because the packing attribute is for gcc only. https://github.com/openzim/libzim/blob/master/src/zim_types.h#L10-L14

I would go with this (and modify the test accordingly).
The dirent size is important for us as it reduce the memory size use by creator (especially for huge zim files) but we don't create zim file on Windows.
And this is not a functional need, if someone create zim on Windows, it will work.